staging/src/k8s.io/kubectl/pkg/cmd/diff/diff.go

```diff
@@ -741,6 +787,40 @@ func (o *DiffOptions) Run() error {
 		return err
 	})
 
+	if o.applySet != nil {
+		validationDirective := "ignore"
+		if err := o.applySet.BeforeApply(infos, cmdutil.DryRunServer, validationDirective); err != nil {
+			return err
+		}
+	}
+
+	if o.applySet != nil {
+		prunedObjects, err := o.applySet.FindObjectsToPrune(ctx, o.DynamicClient, o.tracker.visitedUids)
+		if err != nil {
+			klog.Warningf("pruning failed and could not be evaluated: %v", err)
+		}
+
+		// Print pruned objects into old file and thus, diff
+		// command will show them as pruned.
+		for _, prunedObject := range prunedObjects {
+			obj := prunedObject.Object
+
+			name, err := getObjectName(obj)
+			if err != nil {
+				klog.Warningf("pruning failed and object name could not be retrieved: %v", err)
```

Comment:
This message is also kinda surprising

Reply:
Agreed also; tried to rephrase to be clearer.  Maybe we should return an error because something has gone very wrong here...

Reply:
hm, yeah in both of these cases I'm not sure why we're logging and attempting to continue. Should we go ahead and be more strict in our new version here?

Reply:
Yes, I think so, I'll make the change.  We can always relax it later :-)